middle_end/flambda2/terms/function_params_and_body.rec.ml

```diff
@@ -108,16 +110,28 @@ let print ppf t =
 let params_arity t = t.params_arity
 
 let apply_renaming
-      ({ abst; dbg; params_arity; is_my_closure_used; } as t) perm =
+      ({ abst; dbg; params_arity; is_my_closure_used; free_names_of_body } as t) perm =
   let abst' = A.apply_renaming abst perm in
+  let free_names_of_body' =
+    Or_unknown.map free_names_of_body
+      ~f:(fun free_names_of_body ->
+          Name_occurrences.apply_renaming free_names_of_body perm)
+  in
```

Comment:
I think it would make sense to either move this into the else branch (to avoid the computation when it's not needed) or to check physical equality in the condition below (but that should be redundant: if some of the free names have been renamed, then the body should have changed too.)